packages/lesswrong/components/users/DialogueMatchingPage.tsx

```diff
@@ -353,41 +653,246 @@ const UserTopTags = ({ classes, targetUserId }: { classes: ClassesType, targetUs
+    return (
+      <LWDialog open onClose={onClose}>
+          <DialogTitle>
+            <h2>You submitted, nice job.</h2>
+            <p>This info will be sent to your match partner.</p> 
+            <p>Once they fill out the form, you'll get to see each other's answers and chat about whether a dialogue makes sense.</p>
+          </DialogTitle>
+          <div style={{textAlign: "center"}}>
+            <img style={{maxHeight: "50px"}} src="https://res.cloudinary.com/lesswrong-2-0/image/upload/v1497915096/favicon_lncumn.ico"></img>
+          </div>
+          <DialogActions>
+            <Button onClick={onClose} color="default">
+              Close
+            </Button>
+          </DialogActions>
+      </LWDialog>
+  )}
+
+  return (
+    <LWDialog open onClose={onClose}>
+      <div className={classes.dialogBox}>
+        <DialogTitle className={classes.dialogueTitle}>Alright, you matched with {targetUserDisplayName}!</DialogTitle>
+        <DialogContent >
+          {userSuggestedTopics.length > 0 && <>
+            <div>Here are some topics your match was interested in:</div>
+            <div className={classes.dialogueTopicList}>
+              {userSuggestedTopics.map((topic) => <div className={classes.dialogueTopicRow} key={topic.text}>
+                <Checkbox 
+                    className={classes.dialogueTopicRowTopicCheckbox}
+                    color={topic.matchedPersonPreference === "Yes" ? "primary" : "default"}
+                    checked={topic.preference === "Yes"}
+                    // Set the preference of the topic with the matching text to the new preference
+                    onChange={event => setTopicPreferences(
+                      topicPreferences.map(
+                        existingTopic => existingTopic.text === topic.text ? {
+                          ...existingTopic, 
+                          preference: existingTopic.preference === "Yes" ? "No" as const : "Yes" as const,
+                        } : existingTopic
+                      )
+                    )}
+                  />
+                  <div className={classes.dialogueTopicRowTopicText}>
+                    {topic.text}
+                  </div>
+              </div>)}
+          </div></>}
+            <div>Here are some popular topics on LW. Check any you're interested in discussing.</div>
+            <div className={classes.dialogueTopicList}>
+              {recommendedTopics.map((topic) => <div className={classes.dialogueTopicRow} key={topic.text}>
+                <Checkbox 
+                    className={classes.dialogueTopicRowTopicCheckbox}
+                    color={topic.matchedPersonPreference === "Yes" ? "primary" : "default"}
+                    checked={topic.preference === "Yes"}
+                    // Set the preference of the topic with the matching text to the new preference
+                    onChange={event => setTopicPreferences(
+                      topicPreferences.map(
```

Comment:
This logic should probably be extracted to a named function, and then you just topicPreferences.map(descriptiveNameOfFunction)